spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -83,17 +85,25 @@ public class SimpleJobService implements JobService, DisposableBean {
 
 	private JobOperator jsrJobOperator;
 
+	private AggregateJobQueryDao aggregateJobQueryDao;
+
 	private int shutdownTimeout = DEFAULT_SHUTDOWN_TIMEOUT;
+	String schemaTarget = "boot2";
 
 	public SimpleJobService(SearchableJobInstanceDao jobInstanceDao, SearchableJobExecutionDao jobExecutionDao,
 							SearchableStepExecutionDao stepExecutionDao, JobRepository jobRepository,
-							ExecutionContextDao executionContextDao, JobOperator jsrJobOperator) {
+							ExecutionContextDao executionContextDao, JobOperator jsrJobOperator, AggregateJobQueryDao aggregateJobQueryDao,
+							AppBootSchemaVersion appBootSchemaVersion) {
```

Comment:
The SchemaTarget is known to SimpleJobServiceFactory and should be used my factory to provide all the specifics needed. The SimpleJobService should operate using the provided components.